homeassistant/components/konnected/sensor.py

```diff
@@ -0,0 +1,124 @@
+"""
+Support for DHT and DS18B20 sensors attached to a Konnected device.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.konnected/
+"""
+import logging
+
+from homeassistant.components.konnected import (
+    DOMAIN as KONNECTED_DOMAIN, SIGNAL_DS18B20_NEW, SIGNAL_SENSOR_UPDATE)
+from homeassistant.const import (
+    CONF_DEVICES, CONF_PIN, CONF_TYPE, CONF_NAME, CONF_SENSORS,
+    DEVICE_CLASS_HUMIDITY, DEVICE_CLASS_TEMPERATURE, TEMP_FAHRENHEIT)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+from homeassistant.util.temperature import celsius_to_fahrenheit
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['konnected']
+
+SENSOR_TYPES = {
+    DEVICE_CLASS_TEMPERATURE: ['Temperature', None],
+    DEVICE_CLASS_HUMIDITY: ['Humidity', '%']
+}
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up sensors attached to a Konnected device."""
+    if discovery_info is None:
+        return
+
+    SENSOR_TYPES[DEVICE_CLASS_TEMPERATURE][1] = \
+        hass.config.units.temperature_unit
```

Comment:
We don't need to do this. See below. We should just report the unit that the device is reporting in.